src/autorun.ts

```diff
@@ -50,14 +74,15 @@ export function run<T>(fn: Cb<T>): Observable<T> {
                         context._ = prev_;
                     }
                 }),
+                takeWhile(anyDepRunning, true),
```

Comment:
A similar approach could be taken with errors: `Subject` that `error`s when any of the dependent streams errors.
So that the output stream would be something like
```ts
merge(
  update$,
  errors$
).pipe(
  startWith(void 0),
  switchMap(…),
  distinctUntilChanged(),
  takeUntil(complete$)
)
.subscribe(observer);
```

_Though I'm not sure it'll work, it's just a speculation_

Reply:
You're right about that the expression is running again on completion. That's not good... I'll fix.